File: web-frontend/apps/admin-ui/src/api/core/privilege-department.ts
import { requestClient } from '#/api/request';
import type { PageResult } from '#/api/core/privilege-types';

export interface PrivilegeDepartment {
  id?: string;
  pid?: string;
  parentName?: string;
  name?: string;
  code?: string;
  companyId?: string;
  orderNo?: number;
  status?: number;
  nature?: number;
  createTime?: string;
  updateTime?: string;
  children?: PrivilegeDepartment[];
}

export interface OrganizationTreeVO {
  id?: string;
  name?: string;
  children?: DepartmentTreeVO[];
}

export interface DepartmentTreeVO {
  id?: string;
  companyId?: string;
  name?: string;
  code?: string;
  note?: string;
  pid?: string;
  parentName?: string;
  orderNo?: number;
  leader?: number;
  departmentType?: number;
  status?: number;
  nature?: number;
  createTime?: string;
  children?: DepartmentTreeVO[];
}

export async function getDepartmentPageApi(
  page: number,
  size: number,
  params?: Record<string, any>,
) {
  return requestClient.get<PageResult<PrivilegeDepartment>>(
    '/api/privilege/department/page',
    {
      params: { page, size, ...params },
      responseReturn: 'body',
    },
  );
}

export async function getDepartmentApi(id: string) {
  return requestClient.get<PrivilegeDepartment>(
    `/api/privilege/department/${id}`,
    { responseReturn: 'body' },
  );
}

export async function getDepartmentsByCompanyApi(companyId: string) {
  return requestClient.get<PrivilegeDepartment[]>(
    `/api/privilege/department/company/${companyId}`,
    { responseReturn: 'body' },
  );
}

export async function getDepartmentByCodeApi(code: string) {
  return requestClient.get<PrivilegeDepartment>(
    `/api/privilege/department/code/${code}`,
    { responseReturn: 'body' },
  );
}
export async function getDeptTreeApi(companyId?: string) {
  return requestClient.get<DepartmentTreeVO[]>(
    '/api/privilege/department/tree',
    {
      params: companyId === undefined ? {} : { companyId },
      responseReturn: 'body',
    },
  );
}
export async function getDepartmentTreeApi(companyId?: string) {
  return requestClient.get<OrganizationTreeVO[]>(
    '/api/privilege/department/orgTree',
    {
      params: companyId === undefined ? {} : { companyId },
      responseReturn: 'body',
    },
  );
}

export async function getDepartmentsByPidApi(pid: string) {
  return requestClient.get<PrivilegeDepartment[]>(
    `/api/privilege/department/pid/${pid}`,
    { responseReturn: 'body' },
  );
}

export async function createDepartmentApi(data: Record<string, any>) {
  return requestClient.post('/api/privilege/department', data, {
    responseReturn: 'body',
  });
}

export async function updateDepartmentApi(data: Record<string, any>) {
  return requestClient.put('/api/privilege/department', data, {
    responseReturn: 'body',
  });
}

export async function deleteDepartmentApi(id: string) {
  return requestClient.delete(`/api/privilege/department/${id}`, {
    responseReturn: 'body',
  });
}

File: phoenix-privilege/phoenix-privilege-api/src/main/java/com/phoenix/privilege/entity/PrivilegeDepartment.java
package com.phoenix.privilege.entity;

import com.mybatisflex.annotation.Table;
import com.phoenix.privilege.base.BaseEntity;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@Table("tbl_privilege_department")
public class PrivilegeDepartment extends BaseEntity {

	/** 公司id */
	private String companyId;

	/** 名称 */
	private String name;

	/** 编号 */
	private String code;

	/** 备注 */
	private String note;

	/** 父id */
	private String pid;

	/** 排序号 */
	private Integer orderNo;

	/** 是否是leader1:是 0：不是 */
	private Integer leader;

	/** 是否为自建部门 0-自建部门，1-idm同步部门 */
	private Integer departmentType;

	/** 部门状态，0启用、1禁用 */
	private Integer status;

	/** 部门性质：0部门、1组 */
	private Integer nature;

}
